cmd/provision/main.go

```diff
@@ -276,35 +281,81 @@ func fetchLatestArtefacts(ctx context.Context) (*firmwares, error) {
 
 	klog.Infof("Found latest versions: OS %v, Applet %v", latestOSVer, latestAppletVer)
 
-	fw := &firmwares{
-		BootloaderBlock:       bootloaderBlock,
-		BootloaderConfigBlock: booloaderConfigBlock,
-		TrustedOSBlock:        osBlock,
-		TrustedAppletBlock:    appletBlock,
-	}
+	firmwares := &firmwares{}
 
-	if fw.TrustedOS, err = updateFetcher.GetOS(ctx); err != nil {
+	osFW, err := updateFetcher.GetOS(ctx)
+	if err != nil {
 		return nil, fmt.Errorf("GetOS: %v", err)
 	}
-	klog.Infof("Found OS bundle @ %d", fw.TrustedOS.Index)
+	klog.Infof("Found OS bundle @ %d", osFW.Index)
+	firmwares.trustedOS = &fw{
+		bundle: osFW,
+		block:  osBlock,
+	}
 
-	if fw.TrustedApplet, err = updateFetcher.GetApplet(ctx); err != nil {
+	appletFW, err := updateFetcher.GetApplet(ctx)
+	if err != nil {
 		return nil, fmt.Errorf("GetApplet: %v", err)
 	}
-	klog.Infof("Found Applet bundle @ %d", fw.TrustedApplet.Index)
+	klog.Infof("Found Applet bundle @ %d", appletFW.Index)
+	firmwares.trustedApplet = &fw{
+		bundle: appletFW,
+		block:  appletBlock,
+	}
 
-	if fw.Bootloader, err = updateFetcher.GetBoot(ctx); err != nil {
+	bootFW, err := updateFetcher.GetBoot(ctx)
+	if err != nil {
 		return nil, fmt.Errorf("GetBoot: %v", err)
 	}
-	klog.Infof("Found Bootloader bundle @ %d", fw.Bootloader.Index)
+	klog.Infof("Found Bootloader bundle @ %d", bootFW.Index)
+	firmwares.bootloader = &fw{
+		bundle:      bootFW,
+		block:       bootloaderBlock,
+		bonfigBlock: bootloaderConfigBlock,
+	}
 
-	if fw.Recovery, err = updateFetcher.GetRecovery(ctx); err != nil {
+	recoveryFW, err := updateFetcher.GetRecovery(ctx)
+	if err != nil {
 		return nil, fmt.Errorf("GetRecovery: %v", err)
 	}
-	klog.Infof("Found Recovery bundle @ %d", fw.Recovery.Index)
+	klog.Infof("Found Recovery bundle @ %d", recoveryFW.Index)
+	firmwares.recovery = &fw{
+		bundle: recoveryFW,
+	}
 
 	klog.Info("Loaded firmware artefacts.")
-	return fw, nil
+	return firmwares, nil
+}
+
+func prepareFlashJobs(firmwares *firmwares) (map[string]flashJob, error) {
```

Comment:
Can we swap this `map[string]flashJob` for a new type to reflect the `firmwares` struct? Then the new consts at the top can go away too.